### PR TITLE
feat(kafka-logger): the key field should be optional.

### LINKED ISSUE #2778: DISCUSS(kafka-logger):  the key field should be optional
### Issue description

In the schema of the kafka-logger plugin, the key field is mandatory, as shown below:

```
......
    properties = {
        ......
        key = {type = "string"},
        ......
    },
    required = {"broker_list", "kafka_topic", "key"}
......
```

I think the key field should be optional, because when a message is sent to the Kafka service, the key can be specified or empty. So when the key is empty, it can be used normally.

The key should be optional in the schema：

```
......
    properties = {
        ......
        key = {type = "string"},
        ......
    },
    required = {"broker_list", "kafka_topic"}
......
```

### PR DESCRIPTION
**Pull request by Firstsawyou**
fix #2778

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Review by moonming — CHANGES_REQUESTED** [review]
You need to submit PR after discussing and reaching a consensus. There is none discuss in your thread now

File: doc/zh-cn/plugins/kafka-logger.md [hunk 1]
```diff
@@ -43,7 +43,7 @@
 | ---------------- | ------- | ------ | -------------- | ------- | ------------------------------------------------ |
 | broker_list      | object  | 必须   |                |         | 要推送的 kafka 的 broker 列表。                  |
 | kafka_topic      | string  | 必须   |                |         | 要推送的 topic。                                 |
-| key              | string  | 必须   |                |         | 用于加密消息的密钥。                             |
+| key              | string  | 可选   |                |         | 用于加密消息的密钥。                             |
```
Comment (moonming) [inline doc/zh-cn/plugins/kafka-logger.md:46]: please update doc

**Review by moonming — CHANGES_REQUESTED** [review]

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Comment by Firstsawyou** [comment]
The result of the discussion on the mailing list is: key can be null, that is, the key field is optional.

```
By the way, the definition of message is:
The key is an optional message key that was used for partition assignment.
The key can be null.
https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol
```
Mailing list address：
https://lists.apache.org/thread.html/r9308c1204e6b5f417be5d361a27f84affb46dd206c68a70e317ee2e8%40%3Cdev.apisix.apache.org%3E

File: doc/zh-cn/plugins/kafka-logger.md (hunk 1 above)
Comment (Firstsawyou) [reply doc/zh-cn/plugins/kafka-logger.md]: updated.

**Review by moonming — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Pull request merged**